File: backend/modulos/metas/aplicacion/ListarMetasProyectoUseCase.py
from modulos.metas.dominio.MetaRepositoryPort import MetaRepositoryPort
from modulos.metas.dominio.Entidades import Meta
from typing import List

class ListarMetasProyectoUseCase:
    def __init__(self, repositorio: MetaRepositoryPort):
        self.repositorio = repositorio

    def ejecutar(self, proyecto_id: str, incluir_archivadas: bool = False) -> List[Meta]:
        return self.repositorio.listar_por_proyecto(proyecto_id, incluir_archivadas)

File: frontend/src/modules/proyectos/application/ListarMetasProyectoUseCase.ts
import { Meta } from '../domain/Meta';
import { MetaRepositoryPort } from '../domain/MetaRepositoryPort';

export class ListarMetasProyectoUseCase {
  constructor(private repo: MetaRepositoryPort) {}
  async ejecutar(proyectoId: string): Promise<Meta[]> {
    return this.repo.listar(proyectoId);
  }
}
